crates/e2e-testing/src/spin_controller.rs

```diff
@@ -51,6 +51,14 @@ impl Controller for SpinUp {
             cmd.append(&mut xargs);
         }
 
+        // use the appdir as state dir unless explicitly provided,
+        // in which case it is user's responsibility to clean it up.
+        // this also helps ignore STATE_DIR env variable
+        if !xargs.contains(&"--state_dir") {
```

Comment:
I am doing this here because right now we reuse all our test cases for  `spin up` as well as `spin cloud deploy`, and if I understand correctly, we don't support `state dir` with `spin cloud deploy`.